app-typescript/components/RadioButton.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+
+interface IOption {
+    value: any;
+    label: string;
+    disabled?: boolean;
+    icon?: string;
+}
+
+interface IProps {
+    value: any;
+    options: Array<IOption>;
+    labelSide?: 'left' | 'right';
+    onChange(nextValue: string): void;
+}
+
+export class RadioButton extends React.PureComponent<IProps> {
+    constructor(props: IProps) {
+        super(props);
+        this.handleChange = this.handleChange.bind(this);
```

Comment:
fyi you don't have to bind if you call it via:
```ts
() => this.handleChannge(item.value)
```
only if you would pass it somewhere:
```ts
handleChange(event) {...}

<button onClick={this.handleChange}>
```